blog/0003-how-to-model-aggregates.md

```diff
@@ -0,0 +1,62 @@
+
+Using these techniques and heuristics, we may find aggregates and their limitations in a variety of domains. Yet, since aggregates and their bounds depend on the circumstances and requirements of each domain, there is no precise or standardized way to identify them. Thus it's essential to test out several designs and evaluate them based on things like expressiveness, consistency, performance, scalability, etc.
+
+## How to Design Aggregates and Their Roots?
+
+Aggregates can only be referenced to by the identity of the aggregate root. Aggregates are entities that act as a single point of access and control for the aggregate. Aggregates can only be referenced to by the identity of the aggregate root. The aggregate root is responsible for encapsulating the aggregate's state and behavior, and ensuring its consistency and validity. When designing aggregates and their roots, aspects such as selecting an entity or value object based on its identity, lifetime, and responsibilities as an aggregate root, as well as designing the aggregate root's interface and methods to encapsulate the aggregate's state and behavior, are important considerations. The aggregate root's interface and methods should reflect the language and domain concepts, follow the principle of least astonishment, and the command-query separation principle. It should also design its relationships with other aggregates using references or identifiers.
+
+References or identifiers are simple values that represent the identity of another aggregate, without exposing its internal structure or state. Criteria like expressiveness, consistency, performance, scalability, etc. can be used to design aggregates.
+
+## How to Implement Aggregates and Their Persistence?
+
+Programming languages, frameworks, and data storage technologies are important for implementing aggregates and their durability. Object-oriented programming languages and frameworks can be used to deliver the capabilities and patterns needed for our domain like abstraction, encapsulation, inheritance and polymorphism. Functional programming languages and frameworks can provide features and patterns such as records, unions, pattern matching, partial application, currying, etc. Long-term aggregates may require specific data persistence techniques such as relational databases, document databases, key-value stores, graph databases, event stores, etc. Data persistence strategies are crucial to ensure the integrity and performance of our aggregates. We can create aggregates and their durability using various technologies by taking these options and trade-offs into account.
+
+## Conclusion
```

Comment:
Not sure, but maybe an example can be useful also showcasing the effects/tradeoffs of modelling aggregates in one way or the other.

Reply:
I added a reference to the other blog "how to develop an aggregate"